internal/constellation/helm/overrides.go

```diff
@@ -59,6 +59,12 @@ func extraCiliumValues(provider cloudprovider.Provider, conformanceMode bool, ou
 	extraVals["encryption"] = map[string]any{
 		"strictMode": strictMode,
 	}
+	extraVals["ipMasqAgent"] = map[string]any{
+		"config": map[string]any{
+			"nonMasqueradeCIDRs": []string{output.IPCidrNode},
```

Comment:
Shouldn't this contain the pod network, too?

Reply:
Only on GCP right? There it is enabled through ipv4NativeRoutingCIDR (I manually checked this with tcpdump).

Reply:
Also on the node who has the 10.244.1.0/24 pod cidr I get the following iptables on Azure on the main:
```
root@fedora:/home/cilium# iptables -vL -t nat 
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  114  9844 CILIUM_PRE_nat  all  --  any    any     anywhere             anywhere             /* cilium-feeder: CILIUM_PRE_nat */

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  220 13034 CILIUM_OUTPUT_nat  all  --  any    any     anywhere             anywhere             /* cilium-feeder: CILIUM_OUTPUT_nat */

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  291 19192 CILIUM_POST_nat  all  --  any    any     anywhere             anywhere             /* cilium-feeder: CILIUM_POST_nat */

Chain CILIUM_OUTPUT_nat (1 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain CILIUM_POST_nat (1 references)
 pkts bytes target     prot opt in     out     source               destination         
   73  4627 MASQUERADE  all  --  any    eth0    anywhere            !10.244.1.0/24        /* cilium masquerade non-cluster */
    0     0 ACCEPT     all  --  any    any     anywhere             anywhere             mark match 0xa00/0xe00 /* exclude proxy return traffic from masquerade */
    0     0 SNAT       all  --  any    cilium_host !10.244.1.0/24       !10.244.1.0/24        /* cilium host->cluster masquerade */ to:10.244.1.47
    0     0 SNAT       all  --  any    lxc+    localhost            anywhere             /* cilium host->cluster from 127.0.0.1 masquerade */ to:10.244.1.47

Chain CILIUM_PRE_nat (1 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain KUBE-KUBELET-CANARY (0 references)
 pkts bytes target     prot opt in     out     source               destination     
```